sched/task/task_setcancelstate.c

```diff
@@ -112,7 +112,14 @@ int task_setcancelstate(int state, FAR int *oldstate)
               if ((tcb->flags & TCB_FLAG_TTYPE_MASK) ==
                   TCB_FLAG_TTYPE_PTHREAD)
                 {
+                  tcb->flags &= ~TCB_FLAG_CANCEL_PENDING;
+                  tcb->flags |= TCB_FLAG_CANCEL_DOING;
```

Comment:
why not move TCB_FLAG_CANCEL_DOING/TCB_FLAG_CANCEL_PENDING to tls_info_s? so task_setcancelstate can move libc.